src/generate.js

```diff
@@ -78,20 +78,23 @@ var generate = function(options, pool) {
 
 // Generate a random password.
 self.generate = function(options) {
-	// Set defaults.
-	options = options || {};
-	if (!Object.prototype.hasOwnProperty.call(options, 'length')) options.length = 10;
-	if (!Object.prototype.hasOwnProperty.call(options, 'numbers')) options.numbers = false;
-	if (!Object.prototype.hasOwnProperty.call(options, 'symbols')) options.symbols = false;
-	if (!Object.prototype.hasOwnProperty.call(options, 'exclude')) options.exclude = '';
-	if (!Object.prototype.hasOwnProperty.call(options, 'uppercase')) options.uppercase = true;
-	if (!Object.prototype.hasOwnProperty.call(options, 'lowercase')) options.lowercase = true;
-	if (!Object.prototype.hasOwnProperty.call(options, 'excludeSimilarCharacters')) options.excludeSimilarCharacters = false;
-	if (!Object.prototype.hasOwnProperty.call(options, 'strict')) options.strict = false;
 
-	if (options.strict) {
-		var minStrictLength = 1 + (options.numbers ? 1 : 0) + (options.symbols ? 1 : 0) + (options.uppercase ? 1 : 0);
-		if (minStrictLength > options.length) {
+	var innerOptions = {...options};
+
+	// Set defaults.
+	innerOptions = innerOptions || {};
```

Comment:
Do you want to just change this line to be `options = {...options} || {};`? Then you don't need to rename it to `innerOptions` and most of the diff disappears.